Almars/organizationbranding (#11264)
microsoftgraph/microsoft-graph-docs: api-reference/v1.0/api/organizationalbrandingproperties-create.md

| Application                            | Not supported. |

## HTTP request

A branding is created for an organization, if one does not already exist, using PUT or PATCH.

If branding is already configured, PUT overwrites all existing values regardless of what's in the request body. PATCH only overwrites the values that are included in the request body, leaving the values not included unchanged.

The **id** property is ignored on PUT/PATCH to the /branding singleton. If Content-Language is not specified, the default branding is created, which corresponds to an **id** of `und`. If Content-Language is specified, branding is created for that locale.
<!-- { "blockType": "ignored" } -->

```http
PUT /organization/{id}/branding
PATCH /organization/{id}/branding
```

## Optional query parameters

This method supports the OData query parameters to help customize the response. For general information, see [OData query parameters](/graph/query-parameters).

## Request headers

| Name      |Description|
|:----------|:----------|
| Authorization | Bearer {token}. Required. |
| Content-Type  | application/json. Required.  |
| Content-Language  | Locale. Optional.  |

## Request body

> 

Reviewer: We should have an introductory sentence here, like, "In the request body, include a JSON representation of an **organizationalBrandingProperties** object. The following table lists the required properties."

|:---------------------------------------|:--------------------------------------------|
| Delegated (work or school account)     | Organization.ReadWrite.All |
| Delegated (personal Microsoft account) | Not supported. |
| Application                            | Not supported. |

## HTTP request

A branding is created for an organization, if one does not already exist, using PUT or PATCH.

If branding is already configured, PUT overwrites all existing values regardless of what's in the request body. PATCH only overwrites the values that are included in the request body, leaving the values not included unchanged.

The `id` property is ignored on PUT/PATCH to the /branding singleton. If **Content-Language** is not specified, the default branding is created, which corresponds to an `id` of `und`. If **Content-Language** is specified, branding is created for that locale.
<!-- { "blockType": "ignored" } -->

```http
PUT /organization/{id}/branding
PATCH /organization/{id}/branding
```

## Optional query parameters

This method supports the OData query parameters to help customize the response. For general information, see [OData query parameters](/graph/query-parameters).

> 

Author: The section was removed. Marking as resolved.

---

> ms.author: "almars"

Reviewer 2: I believe we're no longer using/exposing the ms.author property. API doctor will flag its absence as a warning but this might be resolved in the near future. cc @Lauragra can confirm.

Author: Updated

---
title: "Create organizationalBrandingProperties"
description: "Create organization branding."
localization_priority: Normal
author: "kexia"
ms.prod: "microsoft-identity-platform"
doc_type: "apiPageType"
---

# Create organizationalBrandingProperties

> Create an [organizationalBrandingProperties](../resources/organizationalbrandingproperties.md) object. This creates the default branding and optionally a localized branding at the same time. The default branding is loaded when a localized branding set isn't configured for the user's browser language.

Reviewer 3: ```suggestion
Create an [organizationalBrandingProperties](../resources/organizationalbrandingproperties.md) object. This creates the default branding and optionally a localized branding. The default branding is loaded when a localized branding set isn't configured for the user's browser language.
```

## Permissions

One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

| Permission type                        | Permissions (from least to most privileged) |
|:---------------------------------------|:--------------------------------------------|
| Delegated (work or school account)     | Organization.ReadWrite.All |
| Delegated (personal Microsoft account) | Not supported. |
| Application                            | Not supported. |

## HTTP request

A branding is created for an organization, if one does not already exist, using PUT or PATCH.

> If branding is already configured, PUT will overwrite all existing values regardless of what's in the request body. PATCH will only overite the values that are included in the request body, leaving the values not included unchanged.

Reviewer 3: ```suggestion
If branding is already configured, PUT overwrites all existing values regardless of what's in the request body. PATCH only overwrites the values that are included in the request body, leaving the values not included unchanged.
```

The **id** property is ignored on PUT/PATCH to the /branding singleton. If Content-Language is not specified, the default branding is created, which corresponds to an **id** of `und`. If Content-Language is specified, branding is created for that locale.
<!-- { "blockType": "ignored" } -->

```http
PUT /organization/{id}/branding
PATCH /organization/{id}/branding
```

## Optional query parameters

> This method supports some of the OData query parameters to help customize the response. For general information, see [OData query parameters](/graph/query-parameters).

Reviewer 3: If you say that only some of the parameters are supported, you need to list them.

## Request headers

| Name      |Description|
|:----------|:----------|
| Authorization | Bearer {token}. Required. |
| Content-Type  | application/json. Required.  |
| Content-Language  | Locale. Optional.  |

## Request body

| Property     | Type        | Description |
|:-------------|:------------|:------------|

> |backgroundColor|String|Color that will appear in place of the background image in low-bandwidth connections. The primary color of your banner logo or your organization color is recommended to be used here. Specify this in hexadecimal (for example, white is #FFFFFF).|

Reviewer 3: ```suggestion
|backgroundColor|String|Color that appears in place of the background image in low-bandwidth connections. The primary color of your banner logo or your organization color is recommended to be used here. Specify this in hexadecimal (for example, white is #FFFFFF).|
```

> |backgroundImage|Stream|Image that appears as the background of the sign in page. .png or .jpg not larger than 1920x1080 and smaller than 300kb. A smaller image will reduce bandwidth requirements and make page loads more performant.|

Reviewer 3: ```suggestion
|backgroundImage|Stream|Image that appears as the background of the sign-in page. The image is a .png or .jpg that is not larger than 1920x1080 and smaller than 300kb. A smaller image will reduce bandwidth requirements and make page loads more performant.|
```

> |bannerLogo|Stream|A banner version of your company logo which appears appears on the sign-in page. .png or .jpg no larger than 36x245px. We recommend using a transparent image with no padding around the logo.|

Reviewer 3: ```suggestion
|bannerLogo|Stream|A banner version of your company logo that appears on the sign-in page. The banner is a .png or .jpg no larger than 36x245px. We recommend using a transparent image with no padding around the logo.|
```

|signInPageText|String|Text that appears at the bottom of the sign-in box. You can use this to communicate additional information, such as the phone number to your help desk or a legal statement. This text must be Unicode and not exceed 1024 characters.|

> |squareLogo|Stream|Square version of your company logo. This appears in Windows 10 out-of-box (OOBE) experiences and when Windows Autopilot is enabled for deployment. .png or .jpg no larger than 240x240px and no more than 10kb in size. We recommend using a transparent image with no padding around the logo.|

Reviewer 3: ```suggestion
|squareLogo|Stream|Square version of your company logo. This appears in Windows 10 out-of-box (OOBE) experiences and when Windows Autopilot is enabled for deployment. The logo is a .png or .jpg no larger than 240x240px and no more than 10kb in size. We recommend using a transparent image with no padding around the logo.|
```

> |usernameHintText|String|String that shows as the hint in the username textbox on the sign in screen. This text must be Unicode, without links or code, and can't exceed 64 characters.|

Reviewer 3: ```suggestion
|usernameHintText|String|The hint in the username textbox on the sign-in screen. This text must be Unicode, without links or code, and can't exceed 64 characters.|
```

## Response

If successful, this method returns a `201 Created` response code and the created [organizationalBrandingProperties](../resources/organizationalbrandingproperties.md) object in the response body.

## Examples

The following example creates default branding and localization for en-US.

### Request

The following is an example of the request.

> # [HTTP](#tab/http)

Reviewer 3: remove line 73. This comes from an auto-generation of code snippets. It will be automatically added.

<!-- {
  "blockType": "request",
  "name": "get_organizationalbrandingproperties"
}-->

```http
PUT https://graph.microsoft.com/v1.0/organization/d69179bf-f4a4-41a9-a9de-249c0f2efb1d/branding
Content-Type: application/json
Content-Language: en-US

{
    "backgroundColor":"#FFFF33",
    "signInPageText":"Welcome",
    "usernameHintText":"hint"
}
```

> ---

Reviewer 3: Remove line 92.

    "backgroundImage@odata.mediaContentType":"image/*",
    "backgroundImage@odata.mediaReadLink": "https://graph.microsoft.com/v1.0/organization/d69179bf-f4a4-41a9-a9de-249c0f2efb1d/branding/backgroundImage",
    "backgroundImage@odata.mediaEditLink": "https://graph.microsoft.com/v1.0/organization/d69179bf-f4a4-41a9-a9de-249c0f2efb1d/branding/localizations/en-US/backgroundImage",
    "bannerLogo@odata.mediaContentType":"image/*",
    "bannerLogo@odata.mediaReadLink": "https://graph.microsoft.com/v1.0/organization/d69179bf-f4a4-41a9-a9de-249c0f2efb1d/branding/bannerLogo",
    "bannerLogo@odata.mediaEditLink": "https://graph.microsoft.com/v1.0/organization/d69179bf-f4a4-41a9-a9de-249c0f2efb1d/branding/localizations/en-US/bannerLogo",
    "id": "und",
    "signInPageText":"Welcome",
    "squareLogo@odata.mediaContentType":"image/*",
    "squareLogo@odata.mediaReadLink": "https://graph.microsoft.com/v1.0/organization/d69179bf-f4a4-41a9-a9de-249c0f2efb1d/branding/squareLogo",
    "squareLogo@odata.mediaEditLink": "https://graph.microsoft.com/v1.0/organization/d69179bf-f4a4-41a9-a9de-249c0f2efb1d/branding/localizations/en-US/squareLogo",
    "usernameHintText":"hint"
}
```

> In this case, the default branding object is set. Localized branding for en-US is also set due to the Content-Language in the header, even though the en-US branding set is not returned in the response. Note that Content-Language in the request is optional, and if not present, will only set default branding.

Reviewer 3: ```suggestion
In this case, the default branding object is set. Localized branding for en-US is also set due to the Content-Language header, even though the en-US branding set is not returned in the response. The Content-Language header in the request is optional, and if not present, only sets the default branding.
```
